src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -0,0 +1,215 @@
+    } else {
+        // We don't have the opportunity to use a separated queue for data transfer!
+        // Do not create a new queue, use the graphics queue instead.
+        use_distinct_data_transfer_queue = false;
+    }
+
+    if (!use_distinct_data_transfer_queue) {
+        spdlog::warn("The application is forces to avoid distinct data transfer queues.");
+        spdlog::warn("Because of this, the graphics queue will be used for data transfer.");
+
+        transfer_queue_family_index = graphics_queue_family_index;
+    }
+}
+
+Device::Device(const VkInstance &instance, const VkSurfaceKHR &surface, bool enable_vulkan_debug_markers, bool prefer_distinct_transer_queue,
+               const std::optional<std::uint32_t> preferred_physical_device_index)
+    : graphics_card(graphics_card), surface(surface) {
+
+    std::optional<VkPhysicalDevice> selected_graphics_card =
+        settings_decision_maker.decide_which_graphics_card_to_use(instance, surface, preferred_physical_device_index);
+
+    if (!selected_graphics_card.has_value()) {
+        throw std::runtime_error("Error: Could not find suitable graphics card!");
+    }
+
+    graphics_card = selected_graphics_card.value();
+
+    prepare_queues(true);
+
+    std::vector<const char *> device_extensions_wishlist = {
+        // Since we want to draw on a window, we need the swapchain extension.
+        VK_KHR_SWAPCHAIN_EXTENSION_NAME,
+    };
+
+#ifndef NDEBUG
+    if (enable_vulkan_debug_markers) {
+        // Debug markers allow the assignment of internal names to Vulkan resources.
+        // These internal names will conveniently be visible in debuggers like RenderDoc.
+        // Debug markers are only available if RenderDoc is enabled.
+        device_extensions_wishlist.push_back(VK_EXT_DEBUG_MARKER_EXTENSION_NAME);
+    }
+#endif
+
+    std::vector<const char *> enabled_device_extensions;
+
+    for (auto device_extension_name : device_extensions_wishlist) {
+        if (availability_checks.has_device_extension(graphics_card, device_extension_name)) {
+            spdlog::debug("Device extension '{}' is available on this system.", device_extension_name);
+            enabled_device_extensions.push_back(device_extension_name);
+        } else {
+            throw std::runtime_error("Device extension " + std::string(device_extension_name) + " is not available on this system!");
+        }
+    }
+
+    VkPhysicalDeviceFeatures used_features = {};
+
+    // Enable anisotropic filtering.
+    used_features.samplerAnisotropy = VK_TRUE;
+
+    VkDeviceCreateInfo create_info = {};
+    create_info.sType = VK_STRUCTURE_TYPE_DEVICE_CREATE_INFO;
+    create_info.queueCreateInfoCount = static_cast<std::uint32_t>(queues_to_create.size());
+    create_info.pQueueCreateInfos = queues_to_create.data();
+    // Device layers were deprecated in Vulkan some time ago, essentially making all layers instance layers.
+    create_info.enabledLayerCount = 0;
```

Comment:
These don't need to be set, but it depends if we want to be explicit.

Reply:
I would prefer to have them in there to be explicit about it.